boards/samd20_xplained_pro/Cargo.toml

```diff
@@ -0,0 +1,40 @@
+[package]
+name = "samd20_xplained_pro"
+version = "0.9.0"
+authors = ["Kaizen Sparc <kaizenwoof@integralement.nu>"]
+description = "Board Support crate for the Microchip SAMD20 Xplained Pro eval board"
+keywords = ["no-std", "arm", "cortex-m", "embedded-hal"]
+license = "MIT OR Apache-2.0"
+repository = "https://github.com/atsamd-rs/atsamd"
+readme = "README.md"
+edition = "2018"
+
+[dependencies.cortex-m-rt]
+version = "0.7"
+optional = true
+
+[dependencies.atsamd-hal]
+path = "../../hal"
```

Comment:
If I remove the path dependency, it will use the published crate which does not contains the current updates, how should I replace it?
Btw why does including this path makes it a Tier 1 dependency?

Reply:
ah, good point... I think we'll need to leave it as a path dependency for now since it's new.

Tier one BSPs use path dependencies on the HAL, which means for any HAL change affecting BSPs, all Tier one BSPs must be upgraded in the same PR for CI to pass. Tier two BSPs only rely on released versions of the HAL/PACs, so they can be locked to an older version and updated if someone is interested in doing so. This lowers the burden on maintainers as well as folks looking to change the HAL. See [the README](https://github.com/atsamd-rs/atsamd#how-to-use-a-bsp-ie-getting-started-writing-your-own-code) for more details.

Reply:
Ok, I understand now. Maybe we can remove the board implementation from this PR, and I open a new PR with it after this one is merged? (If yes, I would probably keep it until the end of the PR for practical reasons and remove it at the end :))

Reply:
FWIW, to facilitate development, one option is to use a [patch](https://doc.rust-lang.org/cargo/reference/overriding-dependencies.html#the-patch-section) in Cargo.toml .  The idea would be to PR the changes that you want to eventually be published, but don't include the patch section which just refers to your local copy of atsamd-rs instead of the one on crates.io .

That said, it probably still makes sense to publish the PAC and HAL changes, before a separate PR to add the new boards.

Reply:
FWIW, to facilitate development, one option is to use a [patch](https://doc.rust-lang.org/cargo/reference/overriding-dependencies.html#the-patch-section) in Cargo.toml .  The idea would be to PR the changes that you want to eventually be published, but don't include the patch section which just refers to your local copy of atsamd-rs instead of the one on crates.io .

That said, it probably still makes sense to publish the PAC and HAL changes, before a separate PR to add the new boards.